src/App.tsx

```diff
@@ -47,49 +65,97 @@ export function getReorderedGoods(
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  sortType:SortType;
+  isReversed:boolean;
+};
+
+export class App extends React.Component<{}, State> {
+  state:State = {
+    sortType: 0,
+    isReversed: false,
+  };
+
+  sortAlpabet = () => {
+    this.setState({ sortType: 1 });
+  };
+
+  sortLength = () => {
+    this.setState({ sortType: 2 });
+  };
+
+  sortReverse = () => {
+    this.setState(state => ({ isReversed: !state.isReversed }));
+  };
+
+  resetState = () => {
+    this.setState({ sortType: 0, isReversed: false });
+  };
+
+  render() {
+    const { sortType, isReversed } = this.state;
+    const goods = getReorderedGoods(goodsFromServer, { sortType, isReversed });
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            onClick={this.sortAlpabet}
+            className={classNames(
+              'button',
+              'is-info',
+              { 'is-light': sortType !== 1 },
+            )}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-success',
+              { 'is-light': sortType !== 2 },
+            )}
+            onClick={this.sortLength}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-warning',
+              { 'is-light': isReversed === false },
+            )}
+            onClick={this.sortReverse}
+          >
+            Reverse
+          </button>
+
+          {(sortType || isReversed) && (
+            <button
+              type="button"
+              className={classNames(
+                'button',
+                'is-danger',
+                'is-light',
+              )}
+              onClick={this.resetState}
+            >
+              Reset
+            </button>
+          )}
+        </div>
 
-      <ul>
         <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
+          <ul>
+            {goods.map(good => (<li data-cy="Good" key={null}>{good}</li>))}
```

Comment:
```suggestion
            {goods.map(good => (<li data-cy="Good" key={good}>{good}</li>))}
```
good is unique string here, you can use it as a key